### DISCUSSION TITLE
Feathered Edge in ImageMagick.Net

### DISCUSSION THREAD

**Discussion by TheOneAndOnlyKelly** [ImageMagick/ImageMagick#2109] [Help] [answered]
So very new to the ImageMagick community. Trying to use ImageMagick.Net in my project to add a feathered edge to an image that will be overlayed on another image. I really am not sure where to start with this. I've seen commands for, I guess, the command line version like:

convert thumbnail.gif -alpha set -virtual-pixel transparent -channel A \
	-morphology Distance Euclidean:1,10\! +channel feathered.png

but am unsure even how to parse this or convert it to c# code. Any assistance would be appreciated! Thanks!

**Comment by dlemstra** [comment, marked as answer]
Did you use `ImageMagick.NET` or `Magick.NET`? If you mean `Magick.NET` you can use the sample below. Most options are a method  or properties on the `MagickImage` class:

```C#
using (var image = new MagickImage("thumbnail.gif")) // convert thumbnail.gif
{
    // -alpha set
    image.Alpha(AlphaOption.Set);

    // -virtual-pixel transparent
    image.VirtualPixelMethod = VirtualPixelMethod.Transparent;

    // -morphology Distance Euclidean:1,10!
    image.Morphology(MorphologyMethod.Distance, Kernel.Euclidean, "1,10!", Channels.Alpha);

    // no need to use +channel because the channel mask is not preserved and only used in Morphology

    // feathered.png
    image.Write("feathered.png");
}
```

**Reply by TheOneAndOnlyKelly** [reply]
This is the package I got from NuGet:
Magick.NET-Q8-AnyCPU

**Reply by TheOneAndOnlyKelly** [reply]
That's got it, thanks!!

**Reply by dlemstra** [reply]
Happy that you got it resolved. For further quesions please use the discussions in this repository: https://github.com/dlemstra/Magick.NET/discussions instead.

**Reply by TheOneAndOnlyKelly** [reply]
Got it. Sorry, wasn't aware of that site prior
